scripts/rspec-openapi-yaml-to-swagger-ui-json.py

```diff
@@ -0,0 +1,172 @@
+# instructions:
+#   cd path/to/developer-docs
+#   python ./scripts/rspec-openapi-yaml-to-swagger-ui-json.py
+#   bash html
+
+import glob
+from collections import OrderedDict
+
+import yaml
+import os
+import json
+import re
+import ipdb
+
+
+# https://stackoverflow.com/questions/287871/how-do-i-print-colored-text-to-the-terminal
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+cwd = os.getcwd()
+path_to_subject_repo = '../sms-messaging'
+path_to_destination = './source/_static/js/swagger-ui/swagger-initializer.js'
+yaml_paths = glob.glob('{cwd}/{path}/.backstage/*.yaml'.format(cwd=cwd, path=path_to_subject_repo))
+print(yaml_paths)
+warnings_multiple_responses = {}
+warnings_undefined_path_parameter = []
+warnings_filter_sort_page_parameters = []
+warnings_missing_summary = []
+
+
+def process_parameters():
+    # warn if we find any filter/sort/pagination parameters, since they won't serve the documentation well at the
+    # level of individual endpoints
+    if current_path[verb].get('parameters') and \
+            [item['name'] for item in current_path[verb]['parameters'] if
+             re.search(r'(\[\w+])', item['name']) or item['name'] in ['sort', 'page']]:
+        warnings_filter_sort_page_parameters.append(summary)
+
+    # path parameters should be defined once/uniquely across the api, then referenced at each dependent endpoint.
+    # they should appear in the summary path with braces, e.g., /path/to/{my_uuid}/details
+    for parameter in re.findall(r"\{(\w+)}", summary_path):
+
+        # warn if we come across any that aren't already defined up top
+        if parameter not in parameter_references + warnings_undefined_path_parameter:
+            warnings_undefined_path_parameter.append(parameter)
+
+        # add the reference to the shared parameter definition
+        parameter_reference = "#/components/parameters/{}".format(parameter)
+        if parameter_reference not in [pr.items()[0][1] for pr in current_path[verb].setdefault('parameters', [])]:
+            current_path[verb]['parameters'].append({"$ref": parameter_reference})
+
+
+for yaml_path in yaml_paths:
+    with open(yaml_path, 'r') as yaml_file:
+        json_dict = yaml.load(yaml_file.read().rstrip(), Loader=yaml.FullLoader)
+        parameter_references = json_dict.get('components', {}).get('parameters', {}).keys()
+
+        for path in json_dict['paths'].keys():
```

Comment:
No need for keys... looping through a dict will by default give you keys at each iteration.

Reply:
most of the `for` didn't like removing the `.keys()`, due to `RuntimeError: dictionary changed size during iteration`.  something about the `keys()` getting us a live view at the dict gets us past this error.

Reply:
> most of the `for` didn't like removing the `.keys()`, due to `RuntimeError: dictionary changed size during iteration`. something about the `keys()` getting us a live view at the dict gets us past this error.

Yes, that would be an issue if you try to modify the set of keys in the dict during the execution of a  loop.